### PR TITLE
Issue 10 - MariaDB integration

### PR DESCRIPTION
**Pull request by davidbirchwork**
Tested and working. 

Acceptance Test:

- Create a MariaDB 
- set secrets (either through appsettings.json or through environmental variables)
- set the S3 secrets 
    -    set the MariaDB version and Connection String 
    -       (note that the connection string is a single string consisting of server,port,database,username,password)
- upload an asset
- verify it appears in the asset list

### CODE REVIEW

**Comment by senakafdo** [comment]
@davidbirchwork ideally there needs to be a README file for all of the asset services and the acceptance test should be to follow the README and configure these services to use MariaDB. I don't mind if you were planning to address #20 later on, but, ideally, it is better to have that README before anyone attempts to test and validate PRs like this one. Not having the README for a service that is not starting up with default parameters is sort of a blocking problem.

**Comment by oserban** [comment]
I pushed a new config to fix the external volume issue. 

Please note that the docker volume is not deleted when the image is loaded or unloaded. I also tested pruning all the dangling images and network adapters and the volume is not deleted.

**Comment by davidbirchwork** [comment]
@senakafdo documentation is now provided in https://github.com/ove/ove-asset-services/pull/35

File: setup/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      MYSQL_DATABASE: ""
+      MYSQL_USER: ""
+      MYSQL_PASSWORD: ""
+
+  ovehub-ove-asset-manager:
+    image: ovehub/ove-asset-manager:latest
+    depends_on:
+    - ovehub-ove-asset-db
+    ports:
+    - "80:80"
```
Comment (DavidAkroyd) [inline setup/docker-compose.yml]: Does the asset manager run on port 80? This does not seem correct

File: setup/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      ServiceHostUrl: ""
+      MariaDB__ConnectionString: "Server=ovehub-ove-asset-db;Port=3306;Database=DATABASE;User=USER;Password=PASSWORD;"
+      MariaDB__Version: "10.3.10" 
+
+  ovehub-ove-service-imagetiles:
+    image: ovehub/ove-service-imagetiles:latest
+    depends_on:
+    - ovehub-ove-asset-manager
+    ports:
+    - "8090:80"
```
Comment (DavidAkroyd) [inline setup/docker-compose.yml]: Does this mean multiple internal services run on port 80?

File: setup/docker-compose.yml (hunk 1 above)
Comment (oserban) [reply setup/docker-compose.yml]: By default the service runs on port 80. For docker we leave that set to the default and we use docker to remap the port to a different value. In this case the port is remapped to 80 again for  convenience.

File: setup/docker-compose.yml (hunk 2 above)
Comment (oserban) [reply setup/docker-compose.yml]: No. Within a single docker image only one service runs on 80. Both services run in different docker images and the exposed ports are remapped using docker rules.

File: setup/docker-compose.yml (hunk 1 above)
Comment (senakafdo) [reply setup/docker-compose.yml]: @oserban the port mapping (defaults I mean) is a bit concerning. I fixed the same previously, https://github.com/ove/ove-asset-services/pull/23/files and it seems we are back to square one with this. Unless we stick the convention people won't be able to start these out of the box. Can we fix?

File: setup/docker-compose.yml (hunk 2 above)
Comment (senakafdo) [reply setup/docker-compose.yml]: Yet again the ports need to be fixed - FTR, @oserban.

**Comment by senakafdo** [comment]
@davidbirchwork, @oserban, this does not work. I'm getting `fail: Microsoft.EntityFrameworkCore.Database.Connection[20004]
      An error occurred using the connection to database '' on server 'ovehub-ove-asset-db'.
MySql.Data.MySqlClient.MySqlException (0x80004005): Unable to connect to any of the specified MySQL hosts.`.

In my setup/docker-compose.yml, I have MYSQL_DATABASE: "sample", port mapping for MariaDB as 3106:3306, and MariaDB__ConnectionString: "Server=ovehub-ove-asset-db;Port=3106;Database=sample;User=root;Password=Password;".

First question is why doesn't it read the name of the DB at all? Where else do I need to configure the DB details?

**Comment by senakafdo** [comment]
@oserban, you also need to fix the README.md: `Alternatively, you can use docker. Each package has its own docker image, which can be build by executing the build script or docker-compose directly.` This is only the steps to build the docker images. To run the docker images you also need to

```sh
export SERVICE_VERSION=latest
docker-compose -f setup/docker-compose.yml up -d
```

**Pull request merged**